src/main/java/com/epam/aidial/cfg/domain/validator/InterceptorValidator.java

```diff
@@ -43,29 +57,65 @@ private void validateInterceptorName(Interceptor interceptor) {
         }
     }
 
-    private void validateInterceptorRunnerAndEndpoint(Interceptor interceptor) {
+    private void validateInterceptorSource(Interceptor interceptor) {
+        InterceptorSource source = interceptor.getSource();
         String endpoint = interceptor.getEndpoint();
         String configurationEndpoint = interceptor.getConfigurationEndpoint();
-        String interceptorRunner = interceptor.getInterceptorRunner();
 
-        // TODO [VPA]: uncomment validations when FE will support interceptor runners
-        //if (endpoint != null && StringUtils.isBlank(endpoint)) {
-        //    throw new IllegalArgumentException("Invalid endpoint: '%s'".formatted(endpoint));
+        if (source != null) {
+            if (source instanceof InterceptorEndpointsSource) {
+                validateEndpointsSource(endpoint, configurationEndpoint);
+            } else if (source instanceof InterceptorRunnerSource runnerSource) {
+                validateRunnerSource(runnerSource);
+            } else if (source instanceof InterceptorContainerSource containerSource) {
+                validateContainerSource(containerSource);
+            } else {
+                throw new IllegalArgumentException("Unsupported interceptor source: " + source);
+            }
+            return;
+        }
+
+        // TODO [VPA]: uncomment endpoint validations when FE will support interceptor runners
+        //if (endpoint == null) {
+        //    throw new IllegalArgumentException("Missing endpoint and source. At least one of them should be specified");
         //}
 
-        if (configurationEndpoint != null && EndpointValidator.isInvalidUrl(configurationEndpoint)) {
-            throw new IllegalArgumentException("Invalid configuration endpoint: '%s'".formatted(configurationEndpoint));
+        //validateEndpoint(endpoint, COMPLETION_ENDPOINT_LOG_NAME);
+        validateEndpoint(configurationEndpoint, CONFIGURATION_ENDPOINT_LOG_NAME);
```

Comment:
do you need it?

Reply:
why?

Reply:
why do we need separate validation for case when source == null? it should be the same as for `(source instanceof InterceptorEndpointsSource) ` isn't it?